### ISSUE TITLE
Support for push messages from node to node edit dialogue

### ISSUE THREAD

**Issue by Roaders** [node-red/node-red#900]
Hi

I am creating a node for Flic buttons (well, improving the current one actually) and ideally I would like someone to be able to scan for buttons from within Node red rather than having to do it before adding nodes.
This is actually very simple but I would need to be able to send messages from node to the edit dialogue. For example "Button added" or "Hold down button for 7 seconds".
Can I use the existing web socket connection to do this or would I have to create my own?
Are there any suggested ways of displaying these messages within the Edit dialogue?

This is definitely a nice to have feature and not a requirement. It would make setting up Flic buttons a lot easier though.

**Comment by knolleary** [comment]
Right now, you'd have to do this with polling http requests.

We have the `RED.comms` component for pushing messages from the runtime to the editor over the websocket connection, which is used by the node status messages and the Debug panel. But that is not currently part of the published API nodes can use.

We've held back exposing it as a formal API for nodes to use because we didn't want to encourage real-time updates coming to the editor beyond what the Debug panel does. I think it is something worth revisiting - as long as the use cases are there to support doing so.

**Comment by dceejay** [comment]
plus we know debug needs a revamp - so no doubt will change a bit.

**Comment by sbarwe** [comment]
Would be great to have comms in the API. My usecase is like Roaders for configuration feedback of Wireless Sensor Networks

**Comment by knolleary** [comment]
@sbarwe if you want your node's edit dialog to retrieve information from the runtime, you should use the well established pattern of creating an HTTP endpoint that can be called from your node's `oneditprepare`.

You can see a description of the pattern in my answer to this SO question: http://stackoverflow.com/questions/41567175/send-data-on-configuration

**Comment by sbarwe** [comment]
Yes, thats how I do it now - polling 2times per second for changes...

**Comment by knolleary** [comment]
Ok, can you describe your use case a bit more then? Why does it need to be such high frequency whilst the node is being edited? From a user experience point of view, does a 2 second refresh rate (for example) make a huge difference?

**Comment by sbarwe** [comment]
We need to monitor the signalstrength and other values from the sensors in  the setup phase. When they are correct they need to be fixed and  this is the configuration that will be written (like a calibration).

Everything is fine with the current model - it works and configuration should not be done so often.
It is just that it would be great to have comms available  because
1. we can lower the traffic  
2. be more responsive
3. notify from the node runtime on events directly to the editors / nodes instead of aggregating (and let several editors poll)

So from my side, not a must, but nice to have (thought that websockets were just introduced to do that server push thingy aside to the tradional polling).

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
